control-plane/subcommand/sync-catalog/command.go

```diff
@@ -259,6 +276,22 @@ func (c *Command) Run(args []string) int {
 		// it will be the only allowed namespace
 		allowSet = mapset.NewSet(c.flagK8SSourceNamespace)
 	}
+
+	metricsConfig := metrics.SyncCatalogMetricsConfig(c.flagEnableMetrics, c.flagMetricsPort, c.flagMetricsPath)
+	metricsConfig.PrometheusMetricsRetentionTime = c.flagMetricsRetentionTime
+
+	// Create the metrics sink
+	err = c.recordMetrics()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("unable to create metrics sink: %s", err))
+		return 1
+	}
+
+	if c.prometheusSink == nil {
+		c.logger.Error("Prometheus sink not initialized, metrics cannot be displayed")
+		c.prometheusSink = &prometheus.PrometheusSink{}
```

Comment:
I need this to not throw a nil pointer error. Happy to remove

Reply:
will this condition ever not be nil? the default for a pointer field is `nil` and I don't see where we're setting this so I'd assume we always fall into this block right?

Reply:
It had some errors in testing so I put this there, I'll remove it cos it's not a condition I see us hitting in production.

Reply:
oh nah I think it's a valid check, what I mean is there ever a case where this would actually be set? to me it seems like this will always be nil and need to be set

Reply:
No it won't be nil, `recordMetrics()` returns `prometheus.PrometheusSink` which we assign to this field. So `c.prometheusSink` value will only be nil if `recordMetrics()` returns an error(as we just log create metrics sink error and continue running feature).